flang/lib/Lower/Bridge.cpp

```diff
@@ -1326,7 +1326,8 @@ class FirConverter : public Fortran::lower::AbstractConverter {
                                      bool isUnordered) {
     if (isUnordered || sym.has<Fortran::semantics::HostAssocDetails>() ||
         sym.has<Fortran::semantics::UseDetails>()) {
-      if (!shallowLookupSymbol(sym)) {
+      if (!shallowLookupSymbol(sym) &&
+          !sym.test(Fortran::semantics::Symbol::Flag::OmpShared)) {
```

Comment:
Judging by the tests, I don't think we set the privatization flag on loop induction variables. Do you mean that the patch should be rewritten to instead set that flag?

Reply:
We do set the privatization flag on loop induction variables. 
```f90
  !$omp parallel
    !$omp sections
      do i=1,10
      end do
    !$omp end sections
  !$omp end parallel
end
```
```
$ flang-new -fc1 -fdebug-unparse-with-symbols -fopenmp test.f90
!$omp parallel
!$omp sections
!$omp section
 !DEF: /MainProgram1/OtherConstruct1/i (OmpPrivate, OmpPreDetermined) HostAssoc INTEGER(4)
 do i=1,10
 end do
!$omp end sections
!$omp end parallel
end program
```

But I think it's better to use `Symbol::Flag::OmpShared` anyway, since it seems to work and makes the intention more explicit.